packages/react-scripts/template/src/modules/core/modules/redux/components/Redux.jsx

```
@@ -1,29 +1,23 @@
-import { React, PropTypes, Provider } from '../dependencies';
+import { React, PureComponent, PropTypes, Provider } from '../dependencies';
 
-import { createStore } from '../utilities';
+import { configureStore } from '../config';
 
-const Redux = ({ children, initialState, reducers, saga, middlewares }) => {
-    const store = createStore({ initialState, reducers, saga, middlewares });
+class Redux extends PureComponent {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+    };
 
-    return <Provider store={store}>{children}</Provider>;
-};
+    constructor(props) {
+        super(props);
 
-Redux.propTypes = {
-    children: PropTypes.node.isRequired,
-    // eslint-disable-next-line
-    initialState: PropTypes.object,
-    // eslint-disable-next-line
-    reducers: PropTypes.object.isRequired,
-    saga: PropTypes.func,
-    middlewares: PropTypes.arrayOf(PropTypes.func),
-};
+        this.store = configureStore();
+    }
 
-Redux.defaultProps = {
-    initialState: {},
-    *saga() {
-        yield;
-    },
-    middlewares: [],
-};
+    render() {
+        const { store } = this;
+
+        return store ? <Provider store={store}>{this.props.children}</Provider> : null;
```

do you really need to check it?

---

and I think you should at least throw an error if there is no store, but I think it can't happen to have no store here.

---

You're right, there's no point in checking it. I was creating the store initially in `componentDidMount` and forgot to remove the condition.